### DISCUSSION TITLE
How to Query Data and Sort it in descending order using the number?

### DISCUSSION THREAD

**Discussion by JenuelDev** [nuxt/content#1859] [Q&A] [answered]
So I have this list:
![image](https://user-images.githubusercontent.com/31676163/215267257-07c60438-05f3-43df-90b3-427ee162c4eb.png)

And I want to get data and sort it in descending order:
this is wat Im trying to come up but it seems not working.

```js
const myWork = await queryContent("/my-work").sort({ $numeric: -1 }).find();
```

This does not work,, How to make this work? I need help,, thanks thanks :)

**Comment by kokamkarsahil** [comment, marked as answer]
Update:

`$numeric` is boolean and it has to use with field

```js
const myWork = await queryContent("/my-work").sort({ _id: -1, $numeric: true }).find();
```

**Reply by nobkd** [reply]
**UPDATE: The answer above is correct now!**

OLD:
Sorry, I also thought of that solution, but that does only work for files with one-digit numbers used for ordering, because the fields are compared as strings.

So we have 3 files, e.g.: `10.abc.md`, `5.ab.md` and `1.a.md` in the `my-works` directory:

and we order them in reverse using your query, and get this, because they are strings:

```json
[
  {
    "_path": "/my-work/ab",
    "_id": "content:my-work:5.ab.md"
    // ...
  },
  {
    "_path": "/my-work/abc",
    "_id": "content:my-work:10.abc.md"
    // ...
  },
  {
    "_path": "/my-work/a",
    "_id": "content:my-work:1.a.md"
    // ...
  }
]
```

Here's an example that depicts the problem and where you can play with the query: https://stackblitz.com/edit/github-rynn1j?file=app.vue

We have to use the `$numeric` as stated in the question, but for strings that also contain other characters than digits, and I don't know if or how that is possible.

**Reply by kokamkarsahil** [reply]
Yeah, you are right, it only works with 1-digit numbers.
The `$numeric` is working with ContentList component.

```json
[
  {
    "_path": "/my-work/a",
    "_dir": "my-work",
    "_draft": false,
    "_partial": false,
    "_locale": "en",
    "_empty": true,
    "title": "A",
    "description": "",
    "body": {
      "type": "root",
      "children": [],
      "toc": {
        "title": "",
        "searchDepth": 2,
        "depth": 2,
        "links": []
      }
    },
    "_type": "markdown",
    "_id": "content:my-work:1.a.md",
    "_source": "content",
    "_file": "my-work/1.a.md",
    "_extension": "md"
  },
  {
    "_path": "/my-work/ab",
    "_dir": "my-work",
    "_draft": false,
    "_partial": false,
    "_locale": "en",
    "_empty": true,
    "title": "Ab",
    "description": "",
    "body": {
      "type": "root",
      "children": [],
      "toc": {
        "title": "",
        "searchDepth": 2,
        "depth": 2,
        "links": []
      }
    },
    "_type": "markdown",
    "_id": "content:my-work:5.ab.md",
    "_source": "content",
    "_file": "my-work/5.ab.md",
    "_extension": "md"
  },
  {
    "_path": "/my-work/abc",
    "_dir": "my-work",
    "_draft": false,
    "_partial": false,
    "_locale": "en",
    "_empty": true,
    "title": "Abc",
    "description": "",
    "body": {
      "type": "root",
      "children": [],
      "toc": {
        "title": "",
        "searchDepth": 2,
        "depth": 2,
        "links": []
      }
    },
    "_type": "markdown",
    "_id": "content:my-work:10.abc.md",
    "_source": "content",
    "_file": "my-work/10.abc.md",
    "_extension": "md"
  }
]

```
Here is an example: https://stackblitz.com/edit/github-rynn1j-ykvr2f?file=app.vue

**Reply by nobkd** [reply]
Well... it isn't reversed, is it?

:(

Also, your syntax isn't right, it would have to be something like this, to apply the sorting:
```vue
<ContentList path="/my-work" v-slot="{ list }" :sort="{ $numeric: -1 }">
    // ...
</ContentList>
```
But that, like before also puts the entries in a wrong order.

**Reply by kokamkarsahil** [reply]
> Well... it isn't reversed, is it?
> 
> :(
> 
> Also, your syntax isn't right, it would have to be something like this, to apply the sorting:
> 
> ```
> <ContentList path="/my-work" v-slot="{ list }" :sort="{ $numeric: -1 }">
>     // ...
> </ContentList>
> ```
> 
> But that, like before also puts the entries in a wrong order.

Oof, didn't check that but unfortunately, we are both working `$numeric` is boolean and I have corrected that

![image](https://user-images.githubusercontent.com/84327775/215302899-0cf2c045-1c0a-44a9-a834-0565d1d0a5f0.png)


Here you can check that https://stackblitz.com/edit/github-rynn1j-wbb1vr?file=app.vue

**Reply by kokamkarsahil** [reply]
I have also updated my old answer with correct one

**Reply by nobkd** [reply]
That's great!

**Reply by JenuelDev** [reply]
thanks mate,, the `_id` for me works properly,,, without using numeric.. any idea why $numeric need to be added?

**Reply by nobkd** [reply]
As mentioned in https://github.com/nuxt/content/discussions/1859#discussioncomment-4804954 if you don't apply the `$numeric: true`, the ordering works by string sorting, so numbers **longer than one digit** (different than the file names in your image) aren't sorted correctly.

Imagine that you have the numbers `1`, `2`, `10`, `50` as strings.
If you order by string sorting, your sorted string numbers will look like this: `1`, `10`, `2`, `50`

That's not what you expected, so you use the `$numeric` operator, to interpret the digits in a string as if they were numbers, and get the correct order.

**Reply by JenuelDev** [reply]
ah I see,, I get it now.. :)

**Reply by arkaroy135** [reply]
Thanks for the solution. Very helpful.
